### PR TITLE
Mukherjee benchmarks

### PR DESCRIPTION
**Pull request by ivanazuzic**
These benchmarks have been added according to Table 2 from paper [A Thread-Local Semantics and Efficient Static
Analyses for Race Free Programs](https://arxiv.org/pdf/2009.02775.pdf). They are converted to C from [this RATCOP tests](https://bitbucket.org/suvam/ratcop/src/master/src/svcomp15/).

### CODE REVIEW

**Comment by sim642** [comment]
Wouldn't our bench repo be a better place for these?

**Comment by ivanazuzic** [comment]
> Wouldn't our bench repo be a better place for these?

MAybe, but I'll leave them here because Michael sad he'll cherry-pick them to some other place.

**Comment by michael-schwarz** [comment]
Bnechmarks is not really a good word for it. They're more regression tests than anything else. None of them are longer than 100 LoC.

File: tests/regression/36-apron/79-mukherjee_DoubleLock_P3.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// SKIP
```
Comment (michael-schwarz) [inline tests/regression/36-apron/79-mukherjee_DoubleLock_P3.c:1]: Te assert fails when running with GCC.
Are you sure it was `unsigned int count` @ivanazuzic?

**Comment by sim642** [comment]
> Bnechmarks is not really a good word for it. They're more regression tests than anything else. None of them are longer than 100 LoC.

Maybe a separate group for them at least to keep this set separate from our own stuff?

**Comment by michael-schwarz** [comment]
Separate group on top of `use_tids` now.
@ivanazuzic for the problems with the benchmarks that you identified, can you make a PR on top of `use_tids`?

**Comment by michael-schwarz** [comment]
`52/05` must be wrong, the assert there is never reachable?

**Comment by michael-schwarz** [comment]
`52/12` asserts that something does hold and simultaneously does not hold?

**Comment by michael-schwarz** [comment]
`52/06`, `52/09`, `52/10` Mukherjee can also not do according to the table in his own paper.

**Comment by michael-schwarz** [comment]
After annotating according to the table of what Mukherjee can do, we have three tests left where we are worse:

- `"52/03 mukherjee_sssc12", "52/05 mukherjee_simpleLoop"` we report the asserts are unreachable, maybe something is wrong with the benchmark?
- `"52/17 mukherjee_sync01"` has Undefined Behavior

**Comment by michael-schwarz** [comment]
Added on top of `use_tids`.

**Pull request closed without merging**